examples/nlp/token_classification/punctuate_capitalize_infer.py

```diff
@@ -0,0 +1,739 @@
+        setattr(args, default_model_parameter, default_model)
+    for name in ["input_manifest", "input_text", "output_manifest", "output_text", "model_path"]:
+        if getattr(args, name) is not None:
+            setattr(args, name, getattr(args, name).expanduser())
+    return args
+
+
+def load_manifest(manifest: Path):
+    result = []
+    with manifest.open() as f:
+        for i, line in enumerate(f):
+            data = json.loads(line)
+            result.append(data)
+    return result
+
+
+def _get_subtokens_and_subtokens_mask(query: str, tokenizer: TokenizerSpec) -> Tuple[List[str], List[int]]:
+    """
+    Tokenizes input query into subtokens and creates subtokens mask. Subtokens mask is an array of the same length as
+    subtokens array and contains zeros and ones in which. If element of mask equals 1, then corresponding subtoken in
+    subtokens array is first subtoken in some word
+    Args:
+        query: a string that will be tokenized
+        tokenizer: an instance of tokenizer
+    Returns:
+        subtokens: list of subtokens
+        subtokens_mask: list of ints
+    """
+    words = query.strip().split()
+    subtokens = []
+    subtokens_mask = []
+    for j, word in enumerate(words):
+        word_tokens = tokenizer.text_to_tokens(word)
+        subtokens.extend(word_tokens)
+        subtokens_mask.append(1)
+        subtokens_mask.extend([0] * (len(word_tokens) - 1))
+    return subtokens, subtokens_mask
+
+
+def _check_max_seq_length_and_margin_and_step(max_seq_length: int, margin: int, step: int):
+    """
+    Checks values of ``max_seq_length``, ``margin``, and ``step``.
+    Args:
+        max_seq_length: a segment length with ``[CLS]`` and ``[SEP]`` tokens
+        margin: a number of input tokens near edges of segments which are not used in punctuation and capitalization
+            prediction.
+        step: offset of consequent segments.
+    Returns:
+        None
+    """
+    if max_seq_length < 3:
+        raise ValueError(
+            f"Parameter `max_seq_length={max_seq_length}` cannot be less than 3 because `max_seq_length` is a length "
+            f"of a segment with [CLS] and [SEP] tokens."
+        )
+    if margin >= (max_seq_length - 2) // 2 and margin > 0 or margin < 0:
+        raise ValueError(
+            f"Parameter `margin` has to be not negative and less than `(max_seq_length - 2) // 2`. Don't forget about "
+            f"CLS and EOS tokens in the beginning and the end of segment. margin={margin}, "
+            f"max_seq_length={max_seq_length}"
+        )
+    if step <= 0:
+        raise ValueError(f"Parameter `step` has to be positive whereas step={step}")
+    if step > max_seq_length - 2 - 2 * margin:
+        logging.warning(
+            f"Parameter step={step} is too big. It will be reduced to `min(max_seq_length, <maximum query length> + 2) "
+            f"- 2 - 2 * margin`."
+        )
+
+
+def get_features_infer(
```

Comment:
move to the `dataset` class or create a utils file inside https://github.com/NVIDIA/NeMo/tree/main/nemo/collections/nlp/models/token_classification

Reply:
moved to dataset file